### ISSUE TITLE
Resources search result component

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#122]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to create parts of the search result component that is being made in #119. We want the features made in this issue to be conditionally switched to based on the whether `resource` is passed to the aforementioned component's props :)

The designs for this component can be found on the [Components tab in Figma](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=1606%3A585&t=5CR5mIoy4P0OjwjS-1) and can further be seen below:

![Mode=Light, Mobile=False, Type=Resource, Reduced=False](https://github.com/activist-org/activist/assets/24387426/cd30a87f-ffe5-4588-befb-ea4651bfda40)

Mobile:

![Mode=Light, Mobile=True, Type=Resource, Reduced=False](https://github.com/activist-org/activist/assets/24387426/ddcd12d4-8ab6-4ba7-9631-02fdba10525f)

- This issue is blocked by #119 that makes the baseline component

### Contribution

Would be great if someone had interest in working on this! Happy to help, or will get to this myself :)

**Comment by andrewtavis** [comment]
Assigning myself to hold this issue for the team working on it for Code Night 🌙😊

**Comment by anthonyshibitov** [comment]
I’ll take this :)

**Comment by andrewtavis** [comment]
Thanks @anthonyshibitov! Feel free to do whatever icon you see fit for a placeholder resource icon. I’ll get the real one in there in #161 :)

**Comment by andrewtavis** [comment]
@anthonyshibitov, FYI I added the resource icon component via #161 :) You can access that now via `<Icon name="IconResource" />` and also pass arguments like `size`. For coloration you need to use Tailwind's `fill` class property rather than text coloration as works for default icons, but so far it's been working well 😊

Hope all's good with you! :) :)

**Comment by anthonyshibitov** [comment]
@andrewtavis cool cool. i'll get this finished up this week. thanks! :D all's been good, just keeping busy :)

**Comment by andrewtavis** [comment]
Sounds great! We’re planning on doing a backend chat on Saturday, btw :) Would likely be in the morning for you based on timing, so let me know if you’d want to join! Will also write about it in the dev channel on Matrix 🙃

**Comment by andrewtavis** [comment]
The edits in eecea1a close this. I switched some of the icons that I'd chosen before and made sure that coloration is consistent across light and dark mode. I also did some edits to the `ImageEvent` component as I think that having the event image be square as was done in #121 to match the others looks better than how I imagined it in the designs. I think they're all looking really great on the [home screen](https://activist.org/home/) together, @anthonyshibitov! Thanks for all your efforts here 😊
